discovery/targetgroup/targetgroup_test.go

```diff
@@ -17,32 +17,154 @@ import (
 	tests := []struct {
-		json          string
-		expectedReply error
+		json            string
+		expectedReply   error
+		expectedTargets []model.LabelSet
 	}{
 		{
 			json: `	{"labels": {},"targets": []}`,
+			expectedReply:   nil,
+			expectedTargets: []model.LabelSet{},
+		},
+		{
+			json: `	{"labels": {},"targets": ["localhost:9090","localhost:9091"]}`,
 			expectedReply: nil,
+			expectedTargets: []model.LabelSet{
+				model.LabelSet{"__address__": "localhost:9090"},
+				model.LabelSet{"__address__": "localhost:9091"}},
 		},
 		{
 			json: `	{"label": {},"targets": []}`,
-			expectedReply: errors.New("json: unknown field \"label\""),
+			expectedReply:   errors.New("json: unknown field \"label\""),
+			expectedTargets: nil,
 		},
 		{
 			json: `	{"labels": {},"target": []}`,
-			expectedReply: errors.New("json: unknown field \"target\""),
+			expectedReply:   errors.New("json: unknown field \"target\""),
+			expectedTargets: nil,
 		},
 	}
-	tg := Group{}
 
 	for _, test := range tests {
+		tg := Group{}
 		actual := tg.UnmarshalJSON([]byte(test.json))
 		testutil.Equals(t, test.expectedReply, actual)
+		testutil.Equals(t, test.expectedTargets, tg.Targets)
+	}
+
+}
+
+func TestTargetGroupYamlMarshal(t *testing.T) {
+	marshal := func(g interface{}) []byte {
+		d, err := yaml.Marshal(g)
+		if err != nil {
+			panic(err)
+		}
+		return d
+	}
+
+	tests := []struct {
+		expectedYaml  string
+		expectetedErr error
+		group         Group
+	}{
+		{
+			//labels should be omitted if empty
+			group:         Group{},
+			expectedYaml:  "targets: []\n",
+			expectetedErr: nil,
+		},
+		{
+			//targets only exposes addresses
+			group: Group{Targets: []model.LabelSet{
+				model.LabelSet{"__address__": "localhost:9090"},
+				model.LabelSet{"__address__": "localhost:9091"}},
+				Labels: model.LabelSet{"foo": "bar", "bar": "baz"}},
+			expectedYaml:  "targets:\n- localhost:9090\n- localhost:9091\nlabels:\n  bar: baz\n  foo: bar\n",
+			expectetedErr: nil,
+		},
+	}
+
+	for _, test := range tests {
+		actual, err := test.group.MarshalYAML()
+		testutil.Equals(t, test.expectetedErr, err)
+		testutil.Equals(t, test.expectedYaml, string(marshal(actual)))
+	}
+}
+
+func TestTargetGroupYamlUnmarshal(t *testing.T) {
+	unmarshal := func(d []byte) func(interface{}) error {
+		return func(o interface{}) error {
+			return yaml.Unmarshal(d, o)
+		}
+	}
+	tests := []struct {
+		yaml                    string
+		expectedNumberOfTargets int
+		expectedNumberOfLabels  int
+		expectedReply           error
+	}{
+		{
+			//empty targe group
+			yaml:                    "labels:\ntargets:\n",
+			expectedNumberOfTargets: 0,
+			expectedNumberOfLabels:  0,
+			expectedReply:           nil,
+		},
+		{
+			//brackets syntax
+			yaml:                    "labels:\n  my:  label\ntargets:\n  ['localhost:9090', 'localhost:9191']",
+			expectedNumberOfTargets: 2,
+			expectedNumberOfLabels:  1,
+			expectedReply:           nil,
+		},
+		{
+			//hyphen syntax
```

Comment:
We don't need to test every variant of YAML format so I'd remove this test case. It is assumed that the underlying YAML library manages it properly.